### DISCUSSION TITLE
Trailing slashes in URIs

### DISCUSSION THREAD

**Discussion by pfk84** [clue/framework-x#167] [General]
Maybe it makes sense to add info about how to support trailing slashes in routes to the documentation. Or even better on how to make parts optional in general, as documented in FastRoute.

It works by adding `[/]` at the end of the route - a life saver if you have many external links pointing to either version of a path, or a lot of them... :)

**Comment by SimonFrings** [comment]
Hey @pfk84, sounds like a good idea to add this to the documentation.

The same can also be achieved by forwarding the trailing slash route or by using a trailing slash middleware. @clue told me about a middleware he wrote a while back which is currently laying around on his laptop, but we weren't sure if that should be part of Framework X itself or a separate project. This is also part of the reason why I opened up the discussion about [PSR-15 compatibility](https://github.com/clue/framework-x/discussions/185).
